californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -885,7 +885,11 @@ private void handleBlock2Response(final Exchange exchange, final Response respon
 
 		} else if (responseExceedsMaxBodySize(response)) {
 
-			LOGGER.debug("requested resource body exceeds max buffer size [{}], aborting request", maxResourceBodySize);
+			String msg = String.format(
+					"requested resource body [%d bytes] exceeds max buffer size [%d bytes], aborting request",
+					response.getOptions().getSize2(), maxResourceBodySize);
+			LOGGER.debug(msg);
+			exchange.getRequest().setOnResponseError(new IllegalStateException(msg));
 			exchange.getRequest().cancel();
```

Comment:
Here, I choose to do both : `setOnResponseError` and `cancel` to keep backward compatibility.

For our needs, it is important to have the error response event before the cancelled one.

If you think for the 3.0, we should only keep the `setOnResponseError`, we should add a TODO to remember this.

Reply:
OK.

Reply:
OK to add the TODO ?

Reply:
yes.